crates/blockchain/payload.rs

```diff
@@ -277,6 +313,69 @@ impl From<PayloadBuildContext> for PayloadBuildResult {
 }
 
 impl Blockchain {
+    /// Attempts to fetch a payload given it's id. If the payload is still being built, it will be finished.
```

Comment:
Waiting for the current payload to finish is how other clients deal with this? The alternative would be to return the last finished payload (and not risk a missing block).

Reply:
Updated: [873aead]
We now cancel the current rebuild process and return the previous payload if it is requested before time is up